test/simulator/asm/org/apache/cassandra/simulator/asm/ClassTransformer.java

```diff
@@ -137,12 +139,58 @@ private ClassTransformer(int api, ClassWriter classWriter, String className, Enu
         this.methodLogger = MethodLogger.log(api, className);
     }
 
+    public void setUpdateVisibility(boolean updateVisibility)
+    {
+        this.updateVisibility = updateVisibility;
+    }
+
+    /**
+     * Java 11 changed the way that classes defined in the same source file get access to private state (see https://openjdk.org/jeps/181),
+     * rather than trying to adapt to this, this method attempts to make the field/method/class public so that access
+     * is not restricted.
+     */
+    private int makePublic(int access)
+    {
+        if (!updateVisibility)
+            return access;
+        // leave non-user created methods alone
```

Comment:
Not only methods? ACC_SYNTHETIC applies also to fields?

Reply:
updated comment